### PR TITLE
Make `zenml up` prefill username when launching dashboard

### PR DESCRIPTION
**Pull request by strickvl**
`zenml up` launches the dashboard (unless you have a specific env variable set) without any details prefilled for the login screen. This PR addresses that and launches with the `default` preset in the dashboard.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/api.py [hunk 1]
```diff
@@ -46,5 +52,10 @@ def show(ngrok_token: Optional[str] = None) -> None:
         logger.debug(f"Tunneling dashboard from {url} to {ngrok_url}.")
         url = ngrok_url
 
-    url = f"{url}:{port}" if port else url
+    if prefill_username:
+        url = (
+            f"{url}:{port}/login?username={DEFAULT_USERNAME}" if port else url
```
Comment (schustmi) [inline src/zenml/api.py:57]: If someone were to configure their local server with external authentication support, will this mess with it in any way @stefannica?

File: src/zenml/api.py [hunk 2]
```diff
@@ -21,17 +21,23 @@
...
 from zenml.logger import get_logger
+from zenml.zen_stores.base_zen_store import DEFAULT_USERNAME
 
 logger = get_logger(__name__)
 
 
-def show(ngrok_token: Optional[str] = None) -> None:
+def show(
+    ngrok_token: Optional[str] = None,
+    prefill_username: bool = False,
```
[the remark this answers is not in the document]
Comment (strickvl) [reply src/zenml/api.py:30]: No objections from my side, and it's indeed a bit nicer

**Review by fa9r — APPROVED** [review]
LGTM from my side but let's wait for clarification from @stefannica regarding @schustmi's comment

File: src/zenml/api.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/api.py]: A local server with external authentication support is not supported and not even technically possible, not even for development purposes.

**Review by stefannica — APPROVED** [review]

**Pull request merged**